server/app/services/cloud/aws/AwsStorageUtils.java

```diff
@@ -0,0 +1,93 @@
+package services.cloud.aws;
+
+import static com.google.common.base.Preconditions.checkNotNull;
+
+import com.typesafe.config.Config;
+import java.net.URI;
+import java.net.URISyntaxException;
+import java.time.Duration;
+import java.util.concurrent.ExecutionException;
+import software.amazon.awssdk.auth.credentials.AwsCredentials;
+import software.amazon.awssdk.auth.credentials.AwsSessionCredentials;
+import software.amazon.awssdk.regions.Region;
+import software.amazon.awssdk.services.s3.endpoints.S3EndpointProvider;
+
+/** Class providing helper methods for working with AWS Simple Storage Service (S3). */
+public final class AwsStorageUtils {
+  private AwsStorageUtils() {}
+
+  /**
+   * The duration that a signed upload or download request URL from {@link #getSignedUploadRequest}
+   * is valid for use.
+   */
+  public static final Duration AWS_PRESIGNED_URL_DURATION = Duration.ofMinutes(10);
+
+  /**
+   * The path to the config variable containing the endpoint for our local AWS instance
+   * (LocalStack).
+   */
+  public static final String AWS_LOCAL_ENDPOINT_CONF_PATH = "aws.local.endpoint";
+
+  /**
+   * Returns a signed upload request to upload a file with the given {@code fileKey} to the given
+   * {@code bucketName}.
+   */
+  public static SignedS3UploadRequest getSignedUploadRequest(
```

Comment:
**suggestion (blocking):** Recommend making the public interface for this class instance methods and injecting an instance wherever its needed using the default c'tor.

Public static methods are a liability for maintenance and extensibility because if future requirements necessitate access to state (e.g. for consuming the `SettingsManifest` to check a feature flag or config value) the options are to pass that state in at every call state or make them instance methods and refactor every call site to consume an instance.

Reply:
Done.